src/realm/db.cpp

```diff
@@ -1357,6 +1358,153 @@ void DB::open(Replication& repl, const std::string& file, const DBOptions option
+    for (auto col : table->get_column_keys()) {
+        Table* embedded_table = nullptr;
+        if (auto target_table = table->get_opposite_table(col)) {
+            if (target_table->is_embedded())
+                embedded_table = target_table.unchecked_ptr();
+        }
+        cols.emplace_back(col, embedded_table);
+    }
+    return cols;
+}
+
+void generate_properties_for_obj(Replication& repl, const Obj& obj, const ColInfo& cols)
+{
+    for (auto elem : cols) {
+        auto col = elem.first;
+        if (auto embedded_table = elem.second) {
+            // The column contains links to embedded objects
+            auto cols_2 = get_col_info(embedded_table);
+            if (col.is_collection()) {
+                LinkCollectionPtr list = obj.get_linkcollection_ptr(col);
+                CollectionBase& collection_base = dynamic_cast<CollectionBase&>(*list);
+                auto sz = list->size();
+                for (size_t n = 0; n < sz; n++) {
+                    Obj embedded_obj = list->get_object(n);
+                    repl.list_insert(collection_base, n, embedded_obj.get_key(), n);
+                    generate_properties_for_obj(repl, embedded_obj, cols_2);
+                }
+            }
+            else {
+                ObjKey link = obj.get<ObjKey>(col);
+                auto embedded_obj = embedded_table->get_object(link);
+                repl.set(obj.get_table().unchecked_ptr(), col, obj.get_key(), link);
+                generate_properties_for_obj(repl, embedded_obj, cols_2);
+            }
+            continue;
+        }
+        if (col.is_list()) {
+            auto list = obj.get_listbase_ptr(col);
+            auto sz = list->size();
+            for (size_t n = 0; n < sz; n++) {
+                repl.list_insert(*list, n, list->get_any(n), n);
+            }
+        }
+        else if (col.is_set()) {
+            auto set = obj.get_setbase_ptr(col);
+            auto sz = set->size();
+            for (size_t n = 0; n < sz; n++) {
+                repl.set_insert(*set, n, set->get_any(n));
+            }
+        }
+        else if (col.is_dictionary()) {
+            auto dict = obj.get_dictionary(col);
+            size_t n = 0;
+            for (auto [key, value] : dict) {
+                repl.dictionary_insert(dict, n++, key, value);
+            }
+        }
+        else {
+            repl.set(obj.get_table().unchecked_ptr(), col, obj.get_key(), obj.get_any(col));
+        }
+    }
+}
+
+} // namespace
+
+void DB::create_new_history(Replication& repl)
+{
+    Replication* old_repl = get_replication();
+    try {
+        repl.initialize(*this);
+        set_replication(&repl);
+
+        auto tr = start_write();
+        tr->clear_history();
+        std::vector<TableKey> table_keys;
+        for (auto tk : tr->get_table_keys()) {
+            if (tr->table_is_public(tk))
+                table_keys.push_back(tk);
+        }
+        // Create tables
+        for (auto tk : table_keys) {
```

Comment:
Here and elsewhere, we should only track history of public tables which are the ones with a "class_" prefix. Notably, we shouldn't sync the object-store's "metadata" table.

Reply:
Please note that the tables are filtered just above.

Reply:
Oh yes I missed that, apologies!